ProcessLib/BoundaryCondition/NonuniformDirichletBoundaryCondition.cpp

```diff
@@ -0,0 +1,91 @@
+/**
+ * \file
+ *
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "NonuniformDirichletBoundaryCondition.h"
+
+#include "MeshLib/IO/readMeshFromFile.h"
+#include "ProcessLib/Utils/ProcessUtils.h"
+
+namespace ProcessLib
+{
+std::unique_ptr<NonuniformDirichletBoundaryCondition>
+createNonuniformDirichletBoundaryCondition(
+    BaseLib::ConfigTree const& config,
+    NumLib::LocalToGlobalIndexMap const& dof_table, int const variable_id,
+    int const component_id, unsigned const /*integration_order*/,
+    unsigned const /*shapefunction_order*/, MeshLib::Mesh const& bulk_mesh)
+{
+    DBUG("Constructing NonuniformDirichlet BC from config.");
+    //! \ogs_file_param{prj__process_variables__process_variable__boundary_conditions__boundary_condition__type}
+    config.checkConfigParameter("type", "NonuniformDirichlet");
+
+    // TODO handle paths correctly
+    //! \ogs_file_param{prj__process_variables__process_variable__boundary_conditions__boundary_condition__NonuniformDirichlet__mesh}
+    auto const mesh_file = config.getConfigParameter<std::string>("mesh");
+
+    std::unique_ptr<MeshLib::Mesh> boundary_mesh(
+        MeshLib::IO::readMeshFromFile(mesh_file));
+
+    if (!boundary_mesh)
+    {
+        OGS_FATAL("Error reading mesh `%s'", mesh_file.c_str());
+    }
+
+    // Surface mesh and bulk mesh must have equal axial symmetry flags!
+    boundary_mesh->setAxiallySymmetric(bulk_mesh.isAxiallySymmetric());
+
+    // TODO finally use ProcessLib::Parameter here
```

Comment:
Comment can be dropped here. In the Neumann case, one could use either MeshElement or MeshNode, but for Dirichlet, obviously only MeshNode is meaningful.